Classify BreakLines.h

fantasai opened the pull request:
#### fffbee3095b662f425cda421a17c26620e1e363a
<pre>
Classify BreakLines.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=273729">https://bugs.webkit.org/show_bug.cgi?id=273729</a>
<a href="https://rdar.apple.com/127532669">rdar://127532669</a>

Reviewed by Alan Baradlay.

Organize BreakLines.h functions and data into classes to namespace them
and to mark non-public items as private.

* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::findNextBreakablePosition):
* Source/WebCore/rendering/BreakLines.cpp:
(WebCore:::m_table):
* Source/WebCore/rendering/BreakLines.h:
(WebCore::BreakLines::nextBreakablePosition):
(WebCore::BreakLines::LineBreakTable::unsafeLookup const):
(WebCore::BreakLines::isBreakableSpace):
(WebCore::BreakLines::nextBreakableSpace):
(WebCore::BreakLines::nextCharacter):
(WebCore::BreakLines::isBreakable):
(WebCore::nextBreakablePosition): Deleted.
(WebCore::lineBreakTableUnsafeLookup): Deleted.
(WebCore::isBreakableSpace): Deleted.
(WebCore::nextBreakableSpace): Deleted.
(WebCore::nextCharacter): Deleted.
(WebCore::isBreakable): Deleted.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::computePreferredLogicalWidths):
* Source/WebCore/rendering/line/BreakingContext.h:
(WebCore::BreakingContext::handleText):
* Source/WebKitLegacy/ios/Misc/WebUIKitSupport.mm:
(WebKitGetLastLineBreakInBuffer):

Canonical link: <a href="https://commits.webkit.org/278626@main">https://commits.webkit.org/278626@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ffa594330986fbdff6db3dc481ab4f5b4b55e7cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51087 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30389 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3419 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54344 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1776 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53389 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36679 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1451 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41600 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53186 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28019 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44026 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22719 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25344 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1280 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9527 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47323 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1354 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55939 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26197 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1247 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48999 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27445 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44094 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48128 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11188 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28325 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27176 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
If all the methods are static, maybe what you want is `namespace BreakLines {` instead of a new class?

alanbaradlay commented on Source/WebCore/rendering/BreakLines.h:
```diff
@@ -29,45 +29,65 @@
-};
+class BreakLines {
+public:
+    enum class NoBreakSpaceBehavior {
```
nit: maybe use this cleanup to set the underlying type of these scoped enums (normal WebKit practice)?

fantasai replied:
I could set them to something, but since they are only used for templating it actually doesn't matter what they map to... seems like it would just be unnecessary noise?

fantasai committed:
Classify BreakLines.h
https://bugs.webkit.org/show_bug.cgi?id=273729
rdar://127532669

Reviewed by Alan Baradlay.

Organize BreakLines.h functions and data into classes to namespace them
and to mark non-public items as private.

* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::findNextBreakablePosition):
* Source/WebCore/rendering/BreakLines.cpp:
(WebCore:::m_table):
* Source/WebCore/rendering/BreakLines.h:
(WebCore::BreakLines::nextBreakablePosition):
(WebCore::BreakLines::LineBreakTable::unsafeLookup const):
(WebCore::BreakLines::isBreakableSpace):
(WebCore::BreakLines::nextBreakableSpace):
(WebCore::BreakLines::nextCharacter):
(WebCore::BreakLines::isBreakable):
(WebCore::nextBreakablePosition): Deleted.
(WebCore::lineBreakTableUnsafeLookup): Deleted.
(WebCore::isBreakableSpace): Deleted.
(WebCore::nextBreakableSpace): Deleted.
(WebCore::nextCharacter): Deleted.
(WebCore::isBreakable): Deleted.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::computePreferredLogicalWidths):
* Source/WebCore/rendering/line/BreakingContext.h:
(WebCore::BreakingContext::handleText):
* Source/WebKitLegacy/ios/Misc/WebUIKitSupport.mm:
(WebKitGetLastLineBreakInBuffer):

Canonical link: https://commits.webkit.org/278626@main
